javascript/ql/src/semmle/javascript/JsonSchema.qll

```diff
@@ -8,8 +8,8 @@ import javascript
  * Provides classes and predicates for working with JSON schema libraries.
  */
 module JsonSchema {
-  /** A call that validates an input against a JSON schema. */
-  abstract class ValidationCall extends DataFlow::CallNode {
+  /** A node that validates an input against a JSON schema. */
+  abstract class ValidationCall extends DataFlow::Node {
```

Comment:
I don't think we should make `ValidationCall` into a non-call node. I'd rather tweak its member predicates to fit cases where the returned value does not indicate success/failure.

What do you think about:
- Add a predicate for getting an access to a validation error.
- Make `getPolarity()` have no result if the return value does not indicate success/failure.
- Add `DataFlow::Node getAValidationResultAccess(boolean polarity)` which generalizes `getPolarity` to cases where the data-flow node is not the call itself.